crates/re_query_cache2/src/latest_at/to_archetype/bar_chart.rs

```diff
@@ -0,0 +1,65 @@
+// DO NOT EDIT! This file was auto-generated by crates/re_types_builder/src/codegen/rust/to_archetype.rs
+
+#![allow(unused_imports)]
+#![allow(unused_parens)]
+#![allow(clippy::clone_on_copy)]
+
+use crate::CachedLatestAtResults;
+use re_query2::{PromiseResolver, PromiseResult};
+use re_types_core::{Archetype, Loggable as _};
+use std::sync::Arc;
+
+impl crate::ToArchetype<re_types::archetypes::BarChart> for CachedLatestAtResults {
+    #[inline]
+    fn to_archetype(
+        &self,
+        resolver: &PromiseResolver,
+    ) -> PromiseResult<crate::Result<re_types::archetypes::BarChart>> {
+        re_tracing::profile_function!(<re_types::archetypes::BarChart>::name());
+
+        // --- Required ---
+
+        use re_types::components::TensorData;
+        let values = match self.get_required(<TensorData>::name()) {
+            Ok(values) => values,
+            Err(query_err) => return PromiseResult::Ready(Err(query_err)),
+        };
+        let values = match values.to_dense::<TensorData>(resolver) {
+            PromiseResult::Pending => return PromiseResult::Pending,
+            PromiseResult::Error(promise_err) => return PromiseResult::Error(promise_err),
```

Comment:
I'm looking forward to when we can implement [`Try`](https://doc.rust-lang.org/std/ops/trait.Try.html) on `PromiseResult`

Reply:
Yeah, I remembering weeping while writing this...